app/file_sender.py

```diff
@@ -149,22 +151,58 @@ def start_file_sender(readiness_queue):
         logger.info('Successfully connected to SFTP QM directory', sftp_directory=Config.SFTP_QM_DIRECTORY)
     with sftp.SftpUtility(Config.SFTP_PPO_DIRECTORY):
         logger.info('Successfully connected to SFTP PPD directory', sftp_directory=Config.SFTP_PPO_DIRECTORY)
+
+    check_gcp_bucket_ready()
+
     readiness_queue.put(True)
+
     logger.info('Started file sender')
     while True:
         check_partial_files(Config.PARTIAL_FILES_DIRECTORY)
         sleep(Config.FILE_POLLING_DELAY_SECONDS)
 
 
+def check_gcp_bucket_ready():
+    if not Config.SENT_PRINT_FILE_BUCKET:
+        logger.warn('SENT_PRINT_FILE_BUCKET set to empty, skipping uploading files to GCP')
+        return
+
+    try:
+        storage.Client().get_bucket(Config.SENT_PRINT_FILE_BUCKET)
+    except exceptions.GoogleCloudError as exception:
+        logger.error('File upload to GCS failed: {0!s} not fatal, but must be fixed'.format(exception))
```

Comment:
I don't think this message is right, it's not trying to upload anything to the bucket here?
Also you can do
```python
logger.exception('Message string')
```
and it will log the exception stack trace out as well.